### PR TITLE
chore: update all the datadog layer versions

### PR DESCRIPTION
**Pull request by apiarian-datadog**

### CODE REVIEW

File: versions.tf [hunk 1]
```diff
@@ -1,5 +1,5 @@
 terraform {
-  required_version = ">= 1.5.0"
+  required_version = ">= 1.6.0"
```
Comment (duncanpharvey) [inline versions.tf:2]: Looks like the terraform version got caught in the find/replace

File: examples/python/versions.tf [hunk 2]
```diff
@@ -1,5 +1,5 @@
 terraform {
-  required_version = ">= 1.5.0"
+  required_version = ">= 1.6.0"
```
Comment (duncanpharvey) [inline examples/python/versions.tf:2]: Same issue here

File: examples/node/versions.tf [hunk 3]
```diff
@@ -1,5 +1,5 @@
 terraform {
-  required_version = ">= 1.5.0"
+  required_version = ">= 1.6.0"
```
Comment (duncanpharvey) [inline examples/node/versions.tf:2]: Same issue here

File: examples/java/versions.tf [hunk 4]
```diff
@@ -1,5 +1,5 @@
 terraform {
-  required_version = ">= 1.5.0"
+  required_version = ">= 1.6.0"
```
Comment (duncanpharvey) [inline examples/java/versions.tf:2]: Same issue here

File: examples/dotnet/versions.tf [hunk 5]
```diff
@@ -1,5 +1,5 @@
 terraform {
-  required_version = ">= 1.5.0"
+  required_version = ">= 1.6.0"
```
Comment (duncanpharvey) [inline examples/dotnet/versions.tf:2]: Same issue here

**Review by duncanpharvey — CHANGES_REQUESTED** [review]
Looks good with one small fix

File: versions.tf (hunk 1 above)
Comment (apiarian-datadog) [reply versions.tf:2]: aah! thanks!

File: examples/python/versions.tf (hunk 2 above)
Comment (apiarian-datadog) [reply examples/python/versions.tf]: fixed!

File: examples/dotnet/versions.tf (hunk 5 above)
Comment (apiarian-datadog) [reply examples/dotnet/versions.tf]: fixed!

File: examples/node/versions.tf (hunk 3 above)
Comment (apiarian-datadog) [reply examples/node/versions.tf]: fixed!

**Review by duncanpharvey — APPROVED** [review]
LGTM!

File: main.tf [hunk 6]
```diff
@@ -87,7 +87,7 @@ locals {
   }
 
   tags = {
-    dd_sls_terraform_module = "1.5.0"
+    dd_sls_terraform_module = "2.0.0"
```
Comment (apiarian-datadog) [inline main.tf:90]: we're going to 2.0.0 because we're bumping the minimum required aws provider version.

File: versions.tf [hunk 7]
```diff
@@ -4,7 +4,7 @@ terraform {
   required_providers {
     aws = {
       source  = "hashicorp/aws"
-      version = ">= 5.32.0"
+      version = ">= 5.77.0"
```
Comment (apiarian-datadog) [inline versions.tf:7]: this is pushing us to version 2.0.0 for the whole module.

**Review by duncanpharvey — APPROVED** [review]
LGTM!

**Pull request merged**